### PR TITLE
Saml validate token

### PR DESCRIPTION
**Pull request by ChenYan98**

### CODE REVIEW

File: test/Microsoft.IdentityModel.Tokens.Saml.Tests/SamlSecurityTokenHandlerTests.cs [hunk 1]
```diff
@@ -315,6 +316,32 @@ public void ValidateToken(SamlTheoryData theoryData)
             ClaimsPrincipal retVal = null;
             try
             {
+                //if (theoryData.TestId.Contains("IssuerSigningKey set"))
```
Comment (brentschmaltz) [inline test/Microsoft.IdentityModel.Tokens.Saml.Tests/SamlSecurityTokenHandlerTests.cs]: try and use this pattern for creating and using 'context'

https://github.com/AzureAD/azure-activedirectory-identitymodel-extensions-for-dotnet/blob/xml/test/Microsoft.IdentityModel.Tokens.Tests/xml/DelegatingXmlDictionaryReaderTests.cs#L41

Remember to call "TestUtilities.AssertFailIfErrors(context);" outside try catch

Comment (ChenYan98) [reply test/Microsoft.IdentityModel.Tokens.Saml.Tests/SamlSecurityTokenHandlerTests.cs]: Fixed

File: test/Microsoft.IdentityModel.Tokens.Saml.Tests/SamlSecurityTokenHandlerTests.cs [hunk 2]
```diff
@@ -80,6 +79,8 @@ public void CanReadToken(SamlTheoryData theoryData)
             {
                 theoryData.ExpectedException.ProcessException(ex);
```
Comment (brentschmaltz) [inline test/Microsoft.IdentityModel.Tokens.Saml.Tests/SamlSecurityTokenHandlerTests.cs]: You need to pass context to: ProcessNoException AND ProcessException.
ProcessNoException(context), ProcessException(ex, context)

File: test/Microsoft.IdentityModel.Tokens.Saml.Tests/SamlSecurityTokenHandlerTests.cs [hunk 3]
```diff
@@ -185,6 +186,8 @@ public void ReadToken(SamlTheoryData theoryData)
...
```
Comment (brentschmaltz) [inline test/Microsoft.IdentityModel.Tokens.Saml.Tests/SamlSecurityTokenHandlerTests.cs]: You need to pass context to: ProcessNoException AND ProcessException.
ProcessNoException(context), ProcessException(ex, context)

File: test/Microsoft.IdentityModel.Tokens.Saml.Tests/SamlSecurityTokenHandlerTests.cs (hunk 2 above)
Comment (ChenYan98) [reply test/Microsoft.IdentityModel.Tokens.Saml.Tests/SamlSecurityTokenHandlerTests.cs]: Fixed

**Comment by brentschmaltz** [comment]
:shipit:

**Pull request merged**